include/sys/time_units.h

```diff
@@ -11,6 +11,19 @@
 extern "C" {
 #endif
 
+/** @brief System-wide macro to denote "forever" in milliseconds
+ *
+ *  Usage of this macro is limited to APIs that want to expose a timeout value
+ *  that can optionally be unlimited, or "forever".
+ *  This macro can not be fed into kernel functions or macros directly. Use
+ *  @ref SYS_MSEC instead.
+ */
+#define SYS_FOREVER_MS (-1)
+
+/** @brief System-wide macro to convert milliseconds to kernel timeouts
+ */
+#define SYS_MSEC(ms) (ms == SYS_FOREVER_MS ? K_FOREVER : K_MSEC(ms))
```

Comment:
API meeting: @andyross could we instead have `K_MSEC()` take -1 and generate a `k_timeout_t` that is equivalent to `K_FOREVER`?

Reply:
I believe this would be a significant behavior change, since the de-facto semantics of negative timeouts in the kernel is to treat it as "as-soon-as-possible" (zero delay).  Hence:

    #define K_MSEC(ms)      Z_TIMEOUT_MS(ms)
    #define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS(k_ms_to_ticks_ceil32(MAX(t, 0)))

Reply:
on the other hand, adding new macro which is very similar to existing one (SYS_MSEC vs K_MSEC) just creates confusion. Suppose that i'm trying to set timeout to FOREVER in ms, I will use `s32_t t = K_FOREVER`. That does not compile, so i search and find `SYS_FOREVER_MS` and use that and pass it to `K_MSEC` as before and that compiles but result is unexpected, not forever.

Imo, then it would be better to have:
```
#define K_MSEC(t) (t == SYS_FOREVER) ? Z_FOREVER : Z_TIMEOUT_MS(t)
```

Actually, if `K_MSEC` would support that then `SYS_FOREVER` would better fit in `kernel.h` as something like `K_FOREVER_..` defined next to `K_FOREVER` where one is forever representation in time unit and the latter is forever representation used by the kernel.

Reply:
@pabigot @andyross I would very much appreciate feedback on the above proposal. While I agree that adding a new `SYS_MSEC()` is not ideal, I know that @andyross was partial not to mix the `SYS_` and `K_` macros and instead maintaining their independence,